pkg/apis/azure/v1alpha1/types_infrastructure.go

```diff
@@ -187,6 +187,8 @@ type Subnet struct {
 	// Migrated is set when the network layout is migrated from NetworkLayoutSingleSubnet to NetworkLayoutMultipleSubnet.
 	// Only the subnet that was used prior to the migration should have this attribute set.
 	Migrated bool `json:"migrated,omitempty"`
+	// NatGatewayName is enabled
```

Comment:
I would be a bit more descriptive like maybe stating that if its nil then no NATGW is used.

Also you miss the `+optional` tag